documentation/blog/2025-05-20-goose-gets-a-drivers-license/index.md

```diff
@@ -0,0 +1,158 @@
+
+<!-- truncate -->
+
+<YouTubeShortEmbed videoUrl="https://www.youtube.com/embed/QKg2Q6YCzdw" />
+
+Deemkeen's instructions on getting started were pretty clear, and they did a great job of supplying some code to get started, but I wanted to take things a little further.
+
+## Getting Started with MQTT
+
+I needed to get a few tools installed on my development environment, including Docker, MQTT (`brew install mosquitto`), and Java.
+
+A Docker Compose file was provided to get started with MQTT, and I needed to make a few changes, and create some subfolders to store data. Goose helped with these instructions:
+
+```yaml
+version: '3.8'
+
+services:
+  mosquitto:
+    image: eclipse-mosquitto
+    hostname: mosquitto
+    container_name: mosquitto
+    restart: unless-stopped
+    command: /usr/sbin/mosquitto -c /etc/mosquitto/config/mosquitto.conf -v
+    ports:
+      - "0.0.0.0:1883:1883"
+      - "9001:9001"
+    volumes:
+      - ./mosquitto:/etc/mosquitto
+      - ./mosquitto/data:/mosquitto/data
+      - ./mosquitto/log:/mosquitto/log
+```
+
+```sh
+mkdir -p mosquitto/data mosquitto/log mosquitto/config
+```
+
+Then a `docker compose up` command started the MQTT server.
+
+:::info
+By default, this setup will not use authentication for MQTT, but in a production environment, these would be important to set up to avoid unauthorized access to the MQTT server.
+:::
+
+To make sure everything was working, I could run a few commands to test that I could subscribe to a channel on my MQTT Docker container and publish messages to it from another terminal window:
+
+```sh Terminal 1
+# terminal 1: subscribe to a channel called "MBOT/TOPIC"
+mosquitto_sub -h localhost -p 1883 -t MBOT/TOPIC -v
+```
+
+```sh Terminal 2
+# terminal 2: publish a message to the channel "MBOT/TOPIC"
+mosquitto_pub -h localhost -p 1883 -t MBOT/TOPIC -m "BEEP"
+```
+
+We see the resulting message in termainl 1:
+
+```sh
+# terminal 1 sees this output:
+MBOT/TOPIC BEEP
+```
+
+## Setting Up the mbot2
+
+After the assembly of the mbot2 rover, which took about 15 minutes, I used Makeblock's web-based IDE to copy/paste Deemkeen's [Python code](https://github.com/deemkeen/mbotmcp/blob/main/assets/mbot-mqtt.py) to the IDE and upload it to the mbot2. I added appropriate values for wifi, MQTT server, and which MQTT "topic" to subscribe to for commands.
+
+Once the mbot2 rebooted to use the new code, I could reissue the "BEEP" command from my terminal, and the mbot2 beeped. so it was on to the next step.
+
+## Setting up the local MCP server
+
+Being more of a Python developer, I debated asking Goose to rewrite Deemkeen's Java-based MCP server aas a Python application, but I wanted to stick with the original code as much as possible. I was able to get the MCP server compiled by skipping the tests:

```

Comment:
This kinda feels like a distraction from your overall blog post...might be unnecessary.

```suggestion
Being more of a Python developer, I debated asking Goose to rewrite Deemkeen's Java-based MCP server aas a Python application, but I wanted to stick with the original code as much as possible. I was able to get the MCP server compiled by skipping the tests:

```